src/app/Services/UserListService/user-list.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from '@angular/core';
+import { HttpHeaders, HttpClient, HttpParams } from '@angular/common/http';
+import { NotificationService } from '../notification.service';
+import { HOST_URL } from 'src/app/config';
+import { UserListModel } from 'src/app/Models/UserListModel';
+import { UserListFilter } from 'src/app/Models/UserListFilter';
+import { stringify } from '@angular/core/src/render3/util';
+import { filterQueryId } from '@angular/core/src/view/util';
+import { PaginationService } from '../pagination.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserListService {
+
+  url = HOST_URL + '/api/';
+  httpOptions = {
+    headers: new HttpHeaders({
+      'Content-Type': 'application/json'
+    }),
+    params: new HttpParams()
+    .set('page', this.paginationService.pageIndex.toString())
+    .set('pagecount', this.paginationService.userPageSize.toString())
+  };
+
+  constructor(private http: HttpClient, private paginationService: PaginationService) { }
+
+  getUserList() {
+    const typeUrl = this.url + 'userlist/getparams';
+    return this.http.get(typeUrl, this.httpOptions);
+  }
+  getUserListParams(filter: UserListFilter) {
+    const typeUrl = this.url + 'userlist/getparams';
+
+    // if (filter.searchKey != null && filter.searchKey !== '' ) {
+    //   this.httpOptions.params = this.httpOptions.params.set('searchbyname', UserListFilter.name);
+    // } else {
+    //   this.httpOptions.params = this.httpOptions.params.delete('searchbyname');
+    // }
+    // if (filter.selectedStatus != null) {
+    //   this.httpOptions.params = this.httpOptions.params.set('searchbyspecialization', filter.searchKey.toString());
+    // } else {
+    //   this.httpOptions.params = this.httpOptions.params.delete('searchbyspecialization');
+    // }
+    return this.http.get(typeUrl, this.httpOptions);
+  }
+
+  StatusConverter(Users: any) {
+
+    console.log(Users);
+
+    Users.forEach(element => {
+      if (element.statusId === 1 || element.statusName === 'Banned') {
+        element.statusId = 1;
```

Comment:
Whats the logic here? You are checking if its equal 1 and then set it to 1..

Reply:
in future we can post or put user, and this will be easier fill